modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/CommonBeanBaseDefinitionCreator.java

```diff
@@ -120,9 +122,20 @@ private BeanDefinitionBuilder createBeanDefinitionBuilderFromObjectFactory(final
                                                                              final ComponentBuildingDefinition componentBuildingDefinition) {
     Class<?> objectFactoryType = componentBuildingDefinition.getObjectFactoryType();
 
+    if (!enableByteBuddy) {
+      return rootBeanDefinition(objectFactoryClassRepository
+          .getObjectFactoryDynamicClass(componentBuildingDefinition,
+                                        objectFactoryType, componentModel.getType(),
+                                        new LazyValue<>(() -> componentModel.getBeanDefinition().isLazyInit())));
+    }
+
     return rootBeanDefinition(objectFactoryClassRepository
-        .getObjectFactoryClass(componentBuildingDefinition, objectFactoryType, componentModel.getType(),
-                               new LazyValue<>(() -> componentModel.getBeanDefinition().isLazyInit())));
+        .getObjectFactoryClass(objectFactoryType))
+            .addPropertyValue("isSingleton", !componentBuildingDefinition.isPrototype())
+            .addPropertyValue("objectTypeClass", componentModel.getType())
+            .addPropertyValue("isPrototype", componentBuildingDefinition.isPrototype())
+            .addPropertyValue("isEagerInit", new LazyValue<>(() -> !componentModel.getBeanDefinition().isLazyInit()));
```

Comment:
Use the constants from `ObjectFactoryClassRepository` instead of the strings directly.